src/docs/sphinx/dev_guide/updating_tpls.rst

```diff
@@ -0,0 +1,188 @@
+   on a separate branch is a good practice to avoid propagating mis-steps
+   to other users and/or developers.
+
+#. **Changing versions of system packages or other TPLs.**
+   To change a version of a system package, which applies to an LC platforms 
+   or a Docker container image we use for CI testing on GitHub, go into
+   the directory ``axom/scripts/spack/configs``. There you will find a 
+   sub-directory for each supported LC system type. Each sub-directory
+   has a ``packages.yaml`` file which contains an entry for each system level
+   package we rely on. Find the entry for the library you wish to update and 
+   change the version number. Do this for each system you want to test/change,
+   including configurations in the `docker` subdirectory.
+
+   .. note:: Alongside each ``packages.yaml`` in each system package directory,
+             there is a ``compilers.yaml`` file containing compiler and 
+             version information for compilers we use for development and 
+             testing. If you wish to test and build with a new compiler or 
+             version on a system, modify the appropriate ``compilers.yaml`` 
+             file.
+
+   To change a version of a non-system TPL, go into the 
+   ``axom/scripts/spack/packages`` directory. There you will find a 
+   sub-directory for each TPL Axom uses. Modify the contents of the Spack
+   package file ``package.py`` in each package sub-directory as needed. 
+
+#. **Test installation.**
+   Do a test installation of your changes in a local directory by running
+   the following command in the top-level Axom directory::
+
+   $ ./scripts/llnl_scripts/build_tpls.py -d local/install/path
+
+   where ``local/install/path`` is a directory location where you want the 
+   libraries to be installed.
+
+   The script will do a full installation of all TPLs for the plaform you 
+   are running on. It will output whether the TPL install succeeded and, 
+   subsequently, whether an Axom build against the TPL install succeeded.
+
+   Running the script produces new host-config files (i.e., CMake cache files) 
+   that you can use to build and test Axom with the installation, if issues
+   arise. The generated host-config files will be located in the top-level Axom
+   directory of your local clone of the repo. If any changes to Axom code are 
+   needed to work with the TPL update(s), make the changes and test them.
+
+   .. note:: You can build a subset of TPLs for a platform, by passing Spack
+             spec arguments to the Python script. For example,
+
+             ``$ ./scripts/llnl_scripts/build_tpls.py --spec clang@10.0.0~cpp14+devtools+mfem+c2c ...``
+
+             will build the TPLs for the clang 10.0.0 compiler. Please see the
+             ``scripts/spack/specs.json`` file for available specs. 
+
+#. When you are confident that everything is correct, log in as user
+   ``atk`` to each of the machines named in Axom's standard host-configs and run
```

Comment:
Should we add a note that this step has to be done by someone who has access to `atk` ?